Tribler/community/market/core/order.py

```diff
@@ -0,0 +1,384 @@
+        :rtype: bool
+        """
+        return self._is_ask
+
+    @property
+    def cancelled(self):
+        """
+        :return: whether the order has been cancelled or not.
+        :rtype: bool
+        """
+        return self._cancelled
+
+    def is_complete(self):
+        """
+        :return: True if the order is completed.
+        :rtype: bool
+        """
+        return self._traded_quantity >= self._quantity
+
+    @property
+    def status(self):
+        """
+        Return the status of this order. Can be one of these: "open", "completed", "expired" or "cancelled"
+        :return: The status of this order
+        :rtype: str
+        """
+        if self._cancelled:
+            return "cancelled"
+        elif self.is_complete():
+            return "completed"
+        elif self._timeout.is_timed_out(self._timestamp):
+            return "expired"
+        return "open"
+
+    def reserve_quantity_for_tick(self, order_id, quantity):
+        """
+        :param order_id: The order id from another peer that the quantity needs to be reserved for
+        :param quantity: The quantity to reserve
+        :type order_id: OrderId
+        :type quantity: Quantity
+        :return: True if the quantity was reserved, False otherwise
+        :rtype: bool
+        """
+        assert isinstance(order_id, OrderId), type(order_id)
+        assert isinstance(quantity, Quantity), type(quantity)
+
+        if self.available_quantity >= quantity:
+            self._reserved_quantity += quantity
+            if order_id not in self._reserved_ticks:
+                self._reserved_ticks[order_id] = quantity
+            else:
+                self._reserved_ticks[order_id] += quantity
+        else:
+            raise ValueError("Tick does not have enough available quantity for reservation")
+
+        self._logger.debug("reserved quantity for order id %s (own order id: %s),"
+                           "total quantity: %s, traded quantity: %s, reserved quantity: %s",
+                           str(order_id), str(self.order_id), self.total_quantity, self.traded_quantity,
+                           self.reserved_quantity)
+
+    def release_quantity_for_tick(self, order_id, quantity):
+        """
+        Release all quantity for a specific tick.
+        :param order_id: The order id from another peer that the quantity needs to be released for
+        :type order_id: OrderId
+        :raises TickWasNotReserved: Thrown when the tick was not reserved first
+        """
+        if order_id not in self._reserved_ticks:
+            raise TickWasNotReserved()
+
+        if self._reserved_quantity >= quantity:
+            self._reserved_quantity -= quantity
+            self._reserved_ticks[order_id] -= quantity
+            assert self.available_quantity >= Quantity(0, self._quantity.wallet_id), str(self.available_quantity)
+
+            if self._reserved_ticks[order_id] == Quantity(0, quantity.wallet_id):
```

Comment:
Could you add an inline comment here specifying that this is to prevent memory use of 0 Quantity orders? Otherwise future programmers might remove this line.